Maybe uninitialized warning with GCC 13

While compiling Node.js:

```
In function ‘uvwasi__normalize_relative_path’,
    inlined from ‘uvwasi__resolve_path’ at ../../deps/uvwasi/src/path_resolver.c:440:11:
../../deps/uvwasi/src/path_resolver.c:322:5: warning: ‘normalized’ may be used uninitialized [-Wmaybe-uninitialized]
  322 |     uvwasi__free(uvwasi, normalized);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../deps/uvwasi/src/path_resolver.c: In function ‘uvwasi__resolve_path’:
../../deps/uvwasi/src/path_resolver.c:272:9: note: ‘normalized’ was declared here
  272 |   char* normalized;
      |         ^~~~~~~~~~
```

--- a/src/path_resolver.c
+++ b/src/path_resolver.c
@@ -269,7 +269,7 @@ static uvwasi_errno_t uvwasi__normalize_relative_path(
      normalized. */
   uvwasi_errno_t err;
   char* combined;
-  char* normalized;
+  char* normalized = NULL;
   uvwasi_size_t combined_len;
   uvwasi_size_t fd_path_len;
   uvwasi_size_t norm_len;

